### PR TITLE
HIVE-25673: Column pruning fix for MR tasks

### PR DESCRIPTION
**Pull request by pvary**
### What changes were proposed in this pull request?
When updating column pruning information `READ_NESTED_COLUMN_PATH_CONF_STR`, update `READ_COLUMN_NAMES_CONF_STR` and `READ_COLUMN_IDS_CONF_STR` as well. 

### Why are the changes needed?
Iceberg MR queries are failing if multiple tables are queried and several columns are pruned

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Added unit test

### CODE REVIEW

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergSelects.java [hunk 1]
```diff
@@ -203,4 +204,29 @@ public void testScanTableCaseInsensitive() throws IOException {
...
+    Schema schema1 = new Schema(optional(1, "fk", Types.StringType.get()));
+    List<Record> records1 = TestHelper.RecordsBuilder.newInstance(schema1).add("fk1").build();
+    testTables.createTable(shell, "table1", schema1, fileFormat, records1);
+
+    Schema schema2 = new Schema(optional(1, "fk", Types.StringType.get()), optional(2, "val", Types.StringType.get()));
+    List<Record> records2 = TestHelper.RecordsBuilder.newInstance(schema2).add("fk1", "val").build();
+    testTables.createTable(shell, "table2", schema2, fileFormat, records2);
+
+    // MR is needed for the reproduction
+    shell.setHiveSessionValue("hive.execution.engine", "mr");
```
Comment (marton-bod) [inline iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergSelects.java:226]: How's the test coverage for this on Tez? Is it already covered, or do we need to run this for Tez as well to avoid future regressions?

And a minor nit: we're setting the engine to MR but the test output will still show the parameter engine=Tez, which is not a big deal just hurts my OCD a bit :)

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestTables.java [hunk 2]
```diff
@@ -437,7 +437,7 @@ public String identifier(String tableIdentifier) {
       }
 
       Assert.assertTrue(location.delete());
-      return location.toString();
+      return "file://" + location;
```
Comment (marton-bod) [inline iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestTables.java:440]: Why was this needed?

**Review by marton-bod — APPROVED** [review]
LGTM, just couple questions to understand some details better

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergSelects.java (hunk 1 above)
Comment (pvary) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergSelects.java:226]: With Hive 4.0.0 we do not support Iceberg. When I tried to run the tests with MR, the inserts were not working. So I had to run the inserts with Tez, and then the test query with MR.

OTOH this is a valid issue with MR, and older versions on Hive, where MR is supported. (Maybe on newer version as well)

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestTables.java (hunk 2 above)
Comment (pvary) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestTables.java:440]: If the table location does not match the file locations then we had a problem matching the `alias` to the `path`, and failed to add the configuration values to the HiveConf.

We need to use the full path in the table location to match the file names.

File: iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergSelects.java (hunk 1 above)
Comment (marton-bod) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergSelects.java:226]: Got it. By the way, did you mean "With Hive 4.0.0 we do not support MR"?

Comment (pvary) [reply iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergSelects.java:226]: I am not sure that Hive removed MR support (I think not yet, but there is no active development maintaining it), but Hive - Iceberg definitely needs Tez.
